### ISSUE TITLE
CrashLoopBackOff of csi-resizer on k8s 1.15.9-gke.24

### ISSUE THREAD

**Issue by fradeve** [ofek/csi-gcs#35]
Hi all,
I have installed the driver on the k8s version specified in the title, using the following command as per tutorial ```kubectl apply -k "github.com/ofek/csi-gcs/deploy/overlays/stable?ref=master"```.

It resulted in a ```Pod errors: CrashLoopBackOff``` with the following logs:

```
F 2020-05-01T12:41:32.639877Z CSI driver neither supports controller resize nor node resize 
I 2020-05-01T12:46:34.593835Z Version : v0.2.0-0-g07a4fd9 
I 2020-05-01T12:46:34.595055Z Connecting to unix:///var/lib/csi/sockets/pluginproxy/csi.sock 
I 2020-05-01T12:46:34.595785Z Probing CSI driver for readiness 
I 2020-05-01T12:46:34.595807Z GRPC call: /csi.v1.Identity/Probe 
I 2020-05-01T12:46:34.595814Z GRPC request: {} 
I 2020-05-01T12:46:34.597229Z Method Probe called with:  
I 2020-05-01T12:46:34.597298Z Method /csi.v1.Identity/Probe completed 
I 2020-05-01T12:46:34.597913Z GRPC response: {} 
I 2020-05-01T12:46:34.598631Z GRPC error: <nil> 
I 2020-05-01T12:46:34.598650Z GRPC call: /csi.v1.Identity/GetPluginInfo 
I 2020-05-01T12:46:34.598656Z GRPC request: {} 
I 2020-05-01T12:46:34.600401Z Method GetPluginInfo called with:  
I 2020-05-01T12:46:34.600573Z Method /csi.v1.Identity/GetPluginInfo completed 
I 2020-05-01T12:46:34.601053Z GRPC response: {"name":"gcs.csi.ofek.dev","vendor_version":"0.3.0"} 
I 2020-05-01T12:46:34.601726Z GRPC error: <nil> 
I 2020-05-01T12:46:34.601745Z GRPC call: /csi.v1.Identity/GetPluginCapabilities 
I 2020-05-01T12:46:34.601751Z GRPC request: {} 
I 2020-05-01T12:46:34.603106Z Method GetPluginCapabilities called with:  
I 2020-05-01T12:46:34.603151Z Method /csi.v1.Identity/GetPluginCapabilities completed 
I 2020-05-01T12:46:34.603432Z GRPC response: {"capabilities":[{"Type":{"Service":{"type":1}}}]} 
I 2020-05-01T12:46:34.604288Z GRPC error: <nil> 
I 2020-05-01T12:46:34.604307Z GRPC call: /csi.v1.Controller/ControllerGetCapabilities 
I 2020-05-01T12:46:34.604313Z GRPC request: {} 
I 2020-05-01T12:46:34.605020Z Method ControllerGetCapabilities called with: {} 
I 2020-05-01T12:46:34.605699Z Method /csi.v1.Controller/ControllerGetCapabilities completed 
I 2020-05-01T12:46:34.605906Z GRPC response: {"capabilities":[{"Type":{"Rpc":{"type":1}}}]} 
I 2020-05-01T12:46:34.606720Z GRPC error: <nil> 
I 2020-05-01T12:46:34.606838Z GRPC call: /csi.v1.Node/NodeGetCapabilities 
I 2020-05-01T12:46:34.606869Z GRPC request: {} 
I 2020-05-01T12:46:34.607603Z Method NodeGetCapabilities called with: {} 
I 2020-05-01T12:46:34.608171Z Method /csi.v1.Node/NodeGetCapabilities completed 
I 2020-05-01T12:46:34.608383Z GRPC response: {} 
I 2020-05-01T12:46:34.608875Z GRPC error: <nil> 
F 2020-05-01T12:46:34.608902Z CSI driver neither supports controller resize nor node resize 
```
**Update**: after waiting few minutes, the following Pods are up and running correctly:
* csi-gcs-controller
* csi-provisioner

while ```csi-resizer``` is still stuck in a ```CrashLoopBackOff``` with the following log

```
kubectl --namespace kube-system logs csi-gcs-controller-0 csi-resizer
```

```
I0501 12:56:56.591236       1 main.go:61] Version : v0.2.0-0-g07a4fd9
I0501 12:56:56.593028       1 connection.go:151] Connecting to unix:///var/lib/csi/sockets/pluginproxy/csi.sock
I0501 12:56:56.593812       1 common.go:111] Probing CSI driver for readiness
I0501 12:56:56.593845       1 connection.go:180] GRPC call: /csi.v1.Identity/Probe
I0501 12:56:56.593851       1 connection.go:181] GRPC request: {}
I0501 12:56:56.595964       1 connection.go:183] GRPC response: {}
I0501 12:56:56.596671       1 connection.go:184] GRPC error: <nil>
I0501 12:56:56.596799       1 connection.go:180] GRPC call: /csi.v1.Identity/GetPluginInfo
I0501 12:56:56.596873       1 connection.go:181] GRPC request: {}
I0501 12:56:56.599531       1 connection.go:183] GRPC response: {"name":"gcs.csi.ofek.dev","vendor_version":"0.3.0"}
I0501 12:56:56.600027       1 connection.go:184] GRPC error: <nil>
I0501 12:56:56.600148       1 connection.go:180] GRPC call: /csi.v1.Identity/GetPluginCapabilities
I0501 12:56:56.600158       1 connection.go:181] GRPC request: {}
I0501 12:56:56.601132       1 connection.go:183] GRPC response: {"capabilities":[{"Type":{"Service":{"type":1}}}]}
I0501 12:56:56.602091       1 connection.go:184] GRPC error: <nil>
I0501 12:56:56.602109       1 connection.go:180] GRPC call: /csi.v1.Controller/ControllerGetCapabilities
I0501 12:56:56.602115       1 connection.go:181] GRPC request: {}
I0501 12:56:56.603678       1 connection.go:183] GRPC response: {"capabilities":[{"Type":{"Rpc":{"type":1}}}]}
I0501 12:56:56.604463       1 connection.go:184] GRPC error: <nil>
I0501 12:56:56.604580       1 connection.go:180] GRPC call: /csi.v1.Node/NodeGetCapabilities
I0501 12:56:56.604595       1 connection.go:181] GRPC request: {}
I0501 12:56:56.606071       1 connection.go:183] GRPC response: {}
I0501 12:56:56.606709       1 connection.go:184] GRPC error: <nil>
F0501 12:56:56.606721       1 main.go:72] CSI driver neither supports controller resize nor node resize
```

Is there anything that I can do to fix this?

Thanks

**Comment by maennchen** [comment]
@fradeve Sorry, the current `master` definition is not compatible with the released version.

Replace

```
kubectl apply -k "github.com/ofek/csi-gcs/deploy/overlays/stable?ref=master"
```

with

```
kubectl apply -k "github.com/ofek/csi-gcs/deploy/overlays/stable?ref=v0.3.0"
```

**Comment by fradeve** [comment]
Thanks for the quick reply @maennchen, much appreciated! Do you think it is worth documenting this somewhere? I apologise in advance if this was documented somewhere and I missed it :smile:

**Comment by maennchen** [comment]
@fradeve Yet, we should do that. (It was not documented.)

Let's leave this issue open so that we'll update the docs.

**Comment by ofek** [comment]
https://github.com/ofek/csi-gcs/pull/37

@fradeve Thanks!

**Comment by maennchen** [comment]
Replaced by #37

**Comment by fradeve** [comment]
HI @maennchen I just want to double-check something related to your previous suggestion.

After installing v.0.3.0 on GKE 1.15.9, everything worked fine with the exception of some permissions missing from the `csi-gcs-provisioner` ClisterRole.

I have fixed the problem by adding the following rule
```
  - apiGroups: [""]
    resources: ["secrets"]
    verbs: ["get", "list", "watch"]
```
to that ClusterRole. I can see that this rule is present in the same ClusterRole in the `master` branch.

I am just mentioning this here in case you want/need to document this.

**Comment by maennchen** [comment]
@fradeve I'm not sure I understand your question. The rule is present on master as well as on the newest tag.

**Comment by ofek** [comment]
@fradeve I think that was fixed in https://github.com/ofek/csi-gcs/pull/31/files

Should we add `watch` too?

**Comment by maennchen** [comment]
@ofek I don't think we need `watch`....?

**Comment by ofek** [comment]
Just making sure the fix works for them 🙂

**Comment by fradeve** [comment]
> @fradeve I'm not sure I understand your question. The rule is present on master as well as on the newest tag.

Sorry, I should have explained myself a bit better :smile: 

Earlier in this thread you suggested to install v.0.3.0, which worked well. Nonetheless, it got stuck during the deployment process because in v0.3.0 the ClusterRole doesn't allow access to the secret, hence stopping the deployment.

This was indeed fixed by #31 , as pointed out by @ofek. I am not sure `watch` is needed. I wasn't aware of #31 so I thought it could have useful to point this out.

Thanks again!

**Comment by maennchen** [comment]
@fradeve :+1: 

We did a lot of improvements in the latest release. My advice would therefore already change to use `v0.4.0` :)
